### PR TITLE
scripts: sbom: Fix directory interface change in scancode

### PR DESCRIPTION
**Pull request by pdunaj**
Get licenses from the new place.

### CODE REVIEW

**Comment by pdunaj** [comment]
For reference: https://scancode-toolkit.readthedocs.io/en/stable/reference/license-detection-reference.html#change-in-license-data-format-resource

**Comment by thst-nordic** [comment]
the doc says the schema change is implemented in 32.x but we are on 31.2.6
https://github.com/nrfconnect/sdk-nrf/blob/main/scripts/requirements-west-ncs-sbom.txt#L2
do you want to bump to 32+ with this change?

**Comment by pdunaj** [comment]
> the doc says the schema change is implemented in 32.x but we are on 31.2.6 https://github.com/nrfconnect/sdk-nrf/blob/main/scripts/requirements-west-ncs-sbom.txt#L2 do you want to bump to 32+ with this change?

Well spotted - when trying to use sbom I got it failing on the tool missing (seems it was not added by toolchain or west did not pick it up correctly). I have installed it but I have not checked if this is the version we "should" support.

Honestly I tried to do the quick fix and it might be this is not needed at this point.

**Comment by pdunaj** [comment]
checked this with nrf52840 and zephyr's hello_world (one sysbuild domain) and nrf_desktop (3 sysbuild domains). Seems solving the immediate issues.

**Review by thst-nordic — APPROVED** [review]
- sysbuild changes are required
- scancode changes will avoid issue in future

File: scripts/west_commands/sbom/input_build.py [hunk 1]
```diff
@@ -427,9 +434,16 @@ def generate_input(data: Data):
...
+            except:
+                domains = ['.']
+            for domain in domains:
+                domain_build_dir = Path(os.path.join(build_dir, domain))
+                if len(targets) == 0:
+                    targets = [DEFAULT_TARGET]
+                log.dbg(f'INPUT: build directory: {domain_build_dir}, targets: {targets}')
+                b = InputBuild(data, domain_build_dir)
+                for target in targets:
+                    b.generate_from_target(target)
```
Comment (doki-nordic) [inline scripts/west_commands/sbom/input_build.py:449]: In the future, we can consider separating report based on domains, but for now it is ok to put them into one bucket.

**Review by doki-nordic — APPROVED** [review]

**Comment by pdunaj** [comment]
Added exception name to appease the checkpatch

**Comment by pdunaj** [comment]
Rebased

**Pull request merged**